### ISSUE TITLE
Stress Function not work under Grain Module

### ISSUE THREAD

**Issue by KeLiu7** [APSIMInitiative/ApsimX#8032]
### What is your question?

Dear all,

I'm trying to add a stress function under Grain module to affect the final grain number in Next Gen Barley. The original potential grain number is calculated as GrainPerGramOfStem* StemMass.

Now I added a StressWindow to define the timing of stress and the extent of stress value. The final grain number should be calculated as GrainPerGramOfStem* StemMass*Stressvalue, but this seems not work. I also tried to output StressWindow (already under NumberFunction now), it seems the variable is not recognised-nothing relevant with this parameter available in the list...

Does anyone know why this happened? I have attached my Sim file for your reference. Many thanks
![image](https://github.com/APSIMInitiative/ApsimX/assets/72141690/24f4967f-03b3-43f5-928e-497bf63d54b0)

![image](https://github.com/APSIMInitiative/ApsimX/assets/72141690/002e962e-3cc6-45f8-89f3-e9601628ead2)

[Barley_simulation with SWIM3_updated.zip](https://github.com/APSIMInitiative/ApsimX/files/11587564/Barley_simulation.with.SWIM3_updated.zip)

**Comment by KeLiu7** [comment]
@par456 Hi Andre, could you please have a look? many thanks

**Comment by par456** [comment]
I was able to put:

`[Grain].NumberFunction.GrainNumber.StressWindow`

Into the report and have it put out that value during the simulation, I got values of 1.0 and 0.0 across the datastore. The dropdown may just not be working for it since it's something that has been added afterwards.

In regards to your stress not having any impact, I can confirm it is working (since you'll be able to see that value in your datastore), but in your simulation, it changes from 1 to 0 the day after the flowering event happens which is when the NumberFunction stops updating the Grain number. If your stress number was to drop to 0 prior to that, you would see it affecting the Grain Total.

I hope that helps.

**Comment by KeLiu7** [comment]
Thanks Andrew. I think I attached the wrong version. The error says "Cannot find a match for link ttWindowStart in model .Simulation.Field.Barley.Grain.FinalGN.StressFactor.MultiplyFunction.StressWindow”". Could you please check this one? Many thanks
[Barley_simulation with SWIM3_Grain Number.zip](https://github.com/APSIMInitiative/ApsimX/files/11589464/Barley_simulation.with.SWIM3_Grain.Number.zip)

**Comment by par456** [comment]
You'll need to rename your node like the following: 
![image](https://github.com/APSIMInitiative/ApsimX/assets/130029141/16823de5-be3a-4616-af20-0a9baf7841ca)

It seems that StressWindow there needs to have it's nodes be of a certain name in order to work.

That will produce another error afterwards though: Object does not match target type. Which has to do with this part in Phenology: 
![image](https://github.com/APSIMInitiative/ApsimX/assets/130029141/c494eab4-ebbd-42ae-ae15-9d5555eaf8ad)

Basically that variable doesn't exist here: [Phenology].CurrentPhase.ProgressionForTimeStep


But I'll have to leave that up to you, since this is getting back into how you are designing your model. I would recommend looking at something like the Sorghum example for this type of setup, as it has many of the same nodes as what you are doing and will help you figure out what else might be needed.

**Comment by KeLiu7** [comment]
Many thanks Andrew.
